distributed/protocol/tests/test_compression.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
```

Comment:
I want to move all compression-specific tests from test_protocol.py to here.
I'll do it in a successive PR in order to keep code review simple.